### PR TITLE
add 112 path-sum

### PR DESCRIPTION
**Pull request by kobakobashu**
time : 25m

### CODE REVIEW

**Comment by kobakobashu** [comment]
Please check my code.

File: 112.path-sum.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#         self.right = right
+class Solution:
+    def hasPathSum(self, root: TreeNode, targetSum: int) -> bool:
+        """
+        time: O(N)  N : the total of all nodes
+        space: O(N)
+        """
+        if not root:
+            return False
+        if not root.left and not root.right and root.val == targetSum:
```
Comment (ryuji0123) [inline 112.path-sum.py:22]:
```py
if not root.left and not root.right:
    return root.val == targetSum
```

**Review by ryuji0123 — CHANGES_REQUESTED** [review]

**Comment by kobakobashu** [comment]
rebase 忘れた結果，勝手に close してしまいました．直します．

**Comment by kobakobashu** [comment]
I fixed my code.  Please check it.

File: 112.path-sum.py [hunk 2]
```diff
@@ -19,8 +19,10 @@ def hasPathSum(self, root: TreeNode, targetSum: int) -> bool:
         """
         if not root:
             return False
-        if not root.left and not root.right and root.val == targetSum:
-            return True
+
+        if not root.left and not root.right:
+            return root.val== targetSum
```
Comment (ryuji0123) [inline 112.path-sum.py:24]: You don't seem to use flake8 or other linter. I told you to use one.

**Review by ryuji0123 — CHANGES_REQUESTED** [review]

**Comment by kobakobashu** [comment]
I used flake8, and fixed my code.  Please check it.

**Review by ryuji0123 — APPROVED** [review]

**Pull request merged**